### PR TITLE
Validation added

### PR DESCRIPTION
**Pull request by Shubhangi-cs**
Previously the connection was getting tested only on the basis of clientID and Client Secret. Now the validation has been added for all the connection fields.

### CODE REVIEW

File: src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnector.java [hunk 1]
```diff
@@ -119,6 +121,7 @@ public ConnectorSpec generateSpec(ConnectorContext connectorContext, ConnectorSp
     if (template != null) {
       properties.put(AribaPluginConfig.TEMPLATE_NAME, template);
     }
+    properties.put(Constants.Reference.REFERENCE_NAME, ReferenceNames.cleanseReferenceName(template));
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnector.java]: we should not pass arg null to ReferenceNames.cleanseReferenceName()

**Review by rmstar — COMMENTED** [review]
Please update the PR description to provide some context on what you are fixing here and why it's needed

File: src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnector.java (hunk 1 above)
Comment (Shubhangi-cs) [reply src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnector.java]: Reworked on it.

File: src/test/java/io/cdap/plugin/ariba/source/metadata/proto/PropertiesTest.java [hunk 2]
```diff
@@ -238,19 +238,25 @@ public void testGetNonObjectFields() {
   }
 
   @Test
-  public void teatValidateFields() throws AribaException, IOException {
+  public void teatValidateFields() {
```
Comment (rmstar) [inline src/test/java/io/cdap/plugin/ariba/source/metadata/proto/PropertiesTest.java]: teat -> test

File: src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnectorConfig.java [hunk 3]
```diff
@@ -143,36 +141,25 @@ public void validateCredentials(FailureCollector failureCollector) {
...
+      Request req = aribaServices.buildDataRequest(viewTemplatesURL, accessToken);
+      Response response = enhancedOkHttpClient.newCall(req).execute();
+      if (response.code() != HttpURLConnection.HTTP_OK) {
         collector.addFailure("Credentials are incorrect", "Please check the credentials");
       }
-    } catch (IOException e) {
-      LOG.error(" Unable to validate token", e);
+    } catch (IOException | AribaException e) {
+      collector.addFailure("Credentials are incorrect", "Please check the credentials");
+      throw new ValidationException(collector.getValidationFailures());
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnectorConfig.java]: no need to throw exception, failure has been added to the collector.

File: src/test/java/io/cdap/plugin/ariba/source/metadata/proto/PropertiesTest.java [hunk 4]
```diff
@@ -238,19 +238,25 @@ public void testGetNonObjectFields() {
...
     MockFailureCollector collector = new MockFailureCollector();
     AribaConnectorConfig connectorConfig = pluginConfig.getConnection();
-    connectorConfig.validateToken(collector);
-    collector.getOrThrowException();
-    Assert.assertEquals(0, collector.getValidationFailures().size());
+    try {
+      connectorConfig.validateToken(collector);
+      Assert.fail("Validation exception will not be thrown if credentials are correct.");
+    } catch (ValidationException e) {
+      Assert.assertEquals(1, collector.getValidationFailures().size());
```
Comment (rmstar) [inline src/test/java/io/cdap/plugin/ariba/source/metadata/proto/PropertiesTest.java]: Looks like we are only testing validation failure. Can you also add a test for the success case (token validation is successful)?

File: src/test/java/io/cdap/plugin/ariba/source/metadata/proto/PropertiesTest.java (hunk 2 above)
Comment (Shubhangi-cs) [reply src/test/java/io/cdap/plugin/ariba/source/metadata/proto/PropertiesTest.java]: Done

File: src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnectorConfig.java (hunk 3 above)
Comment (Shubhangi-cs) [reply src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnectorConfig.java]: removed

File: src/test/java/io/cdap/plugin/ariba/source/metadata/proto/PropertiesTest.java (hunk 4 above)
Comment (Shubhangi-cs) [reply src/test/java/io/cdap/plugin/ariba/source/metadata/proto/PropertiesTest.java]: Added the test cases for both the types of failures. Positive test case can't be added as the real call for OkHttpClient can’t be mocked due to which we will always get atleast 1  validation failure.

File: src/test/java/io/cdap/plugin/ariba/source/metadata/proto/PropertiesTest.java [hunk 5]
```diff
@@ -238,19 +242,32 @@ public void testGetNonObjectFields() {
...
+      }
+    };
     connectorConfig.validateToken(collector);
-    collector.getOrThrowException();
-    Assert.assertEquals(0, collector.getValidationFailures().size());
+    Assert.assertEquals(1, collector.getValidationFailures().size());
   }
 
-  private void testTest(AribaConnector connector) {
+    private void testTest(AribaConnector connector) {
```
Comment (rmstar) [inline src/test/java/io/cdap/plugin/ariba/source/metadata/proto/PropertiesTest.java]: fix indent

Comment (rmstar) [reply src/test/java/io/cdap/plugin/ariba/source/metadata/proto/PropertiesTest.java]: Also, what is testTest()? Is it a testcase? If yes, annonate with @Test, make it public. Validate that there are no failures

Comment (Shubhangi-cs) [reply src/test/java/io/cdap/plugin/ariba/source/metadata/proto/PropertiesTest.java]: > fix indent

done

Comment (Shubhangi-cs) [reply src/test/java/io/cdap/plugin/ariba/source/metadata/proto/PropertiesTest.java]: > Also, what is testTest()? Is it a testcase? If yes, annonate with @test, make it public. Validate that there are no failures

This is not a testCase, it is being used in another test case that's why @Test annotation is not being added here. Assertion added in the same.

File: src/test/java/io/cdap/plugin/ariba/source/metadata/proto/PropertiesTest.java (hunk 4 above)
Comment (rmstar) [reply src/test/java/io/cdap/plugin/ariba/source/metadata/proto/PropertiesTest.java]: You should test the positive case as well. Mock the OkHttpClient

Comment (Shubhangi-cs) [reply src/test/java/io/cdap/plugin/ariba/source/metadata/proto/PropertiesTest.java]: Done

**Review by rmstar — APPROVED** [review]
please squash commits

**Comment by rmstar** [comment]
Please take a look at e2e test failures

**Pull request merged**